config/helm/aws-node-termination-handler/templates/clusterrole.yaml

```diff
@@ -36,4 +36,3 @@ rules:
     - daemonsets
   verbs:
     - get
```

Comment:
Actually, why does the node termination handler need *any* access to daemonset or replicaset resources? AFAIK, NTH only [queries Node](https://github.com/aws/aws-node-termination-handler/blob/4fe54278740ca7174d50d8f4c7892da55c2219ad/pkg/node/node.go#L313)?

Reply:
Good point, I'll test it out removing those as well and see if there's any gotchas

Reply:
Just a small update, removing the get daemonset permissions is causing the e2e test to break so I'm looking into what's going on there. I'm not totally convinced it's a real error rather than a problem with the test

Reply:
It looks like the daemonset get permissions are required so that we can filter out the daemonsets within the RunNodeDrain call: https://github.com/kubernetes/kubernetes/blob/fb6f5d739b08e049b188806d6e33099f85fad2fe/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L189

I haven't had any issues with removing the replicaset permissions though, so I'll take that out in a new revision.